mitodl/ocw-studio
# Fixed content/file serialization and deserialization

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #226

#### What's this PR do?
Fixes content-to-file transformation and the reverse. `WebsiteContent` records produce the correct file contents when syncing to the storage backend, and those file contents are correctly deserialized into `WebsiteContent` records when re-syncing from the storage backend.

#### How should this be manually tested?
Content to file transformation:
- Reset all `ContentSyncState` objects for a website (i.e.: set `synced_checksum` to null)
- Make sure you have `content_filepath` set for all `WebsiteContent` objects you want synced (this behavior will be fixed in a later issue)
- Sync all contents for that website to Github (see below code snippets)
- Navigate to your Github repo and make sure all files are at the expected locations with the expected contents

File to content transformation:
- Delete some or all of the `WebsiteContent` objects associated with the files that were created. You can also change some of the field values.
- Sync all files to website contents (see below code snippets)
- Inspect the relevant `WebsiteContent` objects in Django admin. Whether the objects were deleted or changed, those records should reflect the data in the files on Github

#### Any background context you want to provide?

Here are some helper scripts since we don't yet have management commands to perform these actions. Run these in a Django shell:

**Sync content to Github**
```python
WEBSITE_NAME = "<your_website_name>"
#
from content_sync.tasks import sync_website_content
sync_website_content.delay(WEBSITE_NAME)
```

**Sync Github files to database**
```python
WEBSITE_NAME = "<your_website_name>"
#
from websites.models import Website, WebsiteContent 
from content_sync.backends.github import GithubBackend
website = Website.objects.get(name="gavin-test-site")
github_backend = GithubBackend(website)
github_backend.sync_all_content_to_db()
```
